test/denormalized_has_many_test.rb

```diff
@@ -12,7 +12,23 @@ def setup
     @record.reload
   end
 
+  def test_uncached_record_from_the_db_will_use_normal_association_as_ary
+    # Test configuration where fetch association returns an array
+    IdentityCache.allow_fetch_to_return_array do    
+      assert_equal IdentityCache.fetch_returns_relation, false
+      expected = @record.associated_records.to_a
+      record_from_db = Item.find(@record.id)
+      assert_queries(1) do
+        assert_memcache_operations(0) do
+          assert_equal expected, record_from_db.fetch_associated_records
```

Comment:
It looks like a collection proxy implements `==` to compare the loaded target to the other object, so I don't think this will actually catch a regression.  I think the assertion you actually want is `assert_equal Array, @record.fetch_associated_records.class`